### PR TITLE
Add FactoryMedia

### PR DESCRIPTION
**Pull request by xiaopengli89**

### CODE REVIEW

File: src/dxgi.rs [hunk 1]
```diff
@@ -73,6 +74,7 @@ crate::weak_com_inheritance_chain! {
         Factory4(dxgi1_4::IDXGIFactory4), from_factory4, as_factory4, unwrap_factory4;
         Factory5(dxgi1_5::IDXGIFactory5), from_factory5, as_factory5, unwrap_factory5;
         Factory6(dxgi1_6::IDXGIFactory6), from_factory6, as_factory6, unwrap_factory6;
+        FactoryMedia(dxgi1_3::IDXGIFactoryMedia), from_factory_media, as_factory_media, unwrap_factory_media;
```
Comment (cwfitzgerald) [inline src/dxgi.rs:77]: This doesn't do what you think it does. This means that if `dxgi1_3::IDXGIFactoryMedia` than all previous versions must also exist. There's no guarantee that dxgi1_6 is supported when IDXGIFactoryMedia is

Comment (xiaopengli89) [reply src/dxgi.rs:77]: Ok, let me fix it.

Comment (xiaopengli89) [reply src/dxgi.rs]: It looks like [IDXGIFactoryMedia](https://learn.microsoft.com/en-us/windows/win32/api/dxgi1_3/nn-dxgi1_3-idxgifactorymedia) inherits directly from [IUnknown](https://learn.microsoft.com/en-us/windows/desktop/api/unknwn/nn-unknwn-iunknown).

**Comment by cwfitzgerald** [comment]
Sorry about the horribly long delay

**Pull request merged**